instrumentation/log4j/log4j-appender-2.16/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_16/internal/LogEventMapper.java

```diff
@@ -76,17 +86,22 @@ public void mapLogEvent(
       @Nullable Instant timestamp,
       T contextData) {
 
+    AttributesBuilder attributes = Attributes.builder();
+
     if (message != null) {
-      builder.setBody(message.getFormattedMessage());
+      if (message instanceof MapMessage) {
+        builder.setBody(message.getFormat());
```

Comment:
Weird API :O

Looks like `getFormat` returns the message for `StructuredDataMessage` and empty string for `StringMapMessage`. From what I've seen, it's conventional to use a map message with an attribute `message` for the body. How about doing `instanceof StructuredDataMessage` for `message.getFormat()`, and checking the attribute for StringMapMessage, leaving Body empty if it's not set?